process/coordinator/process.go

```diff
@@ -981,6 +985,79 @@ func (tc *transactionCoordinator) CreateMarshalizedReceipts() ([]byte, error) {
 	return tc.marshalizer.Marshal(receiptsBatch)
 }
 
+func (tc *transactionCoordinator) getNumOfCrossInterMbsAndTxs() (int, int) {
+	totalNumMbs := 0
+	totalNumTxs := 0
+
+	for _, blockType := range tc.keysInterimProcs {
+		interimProc := tc.getInterimProcessor(blockType)
+		if check.IfNil(interimProc) {
+			continue
+		}
+
+		numMbs, numTxs := interimProc.GetNumOfCrossInterMbsAndTxs()
+		totalNumMbs += numMbs
+		totalNumTxs += numTxs
+	}
+
+	return totalNumMbs, totalNumTxs
+}
+
+func (tc *transactionCoordinator) isMaxBlockSizeReached(body *block.Body) bool {
+	numMbs := len(body.MiniBlocks)
+	numTxs := 0
+	numCrossShardScCalls := 0
+
+	allTxs := make(map[string]data.TransactionHandler)
+	for _, blockType := range tc.keysInterimProcs {
+		txs := tc.GetAllCurrentUsedTxs(blockType)
+		for txHash, tx := range txs {
+			allTxs[txHash] = tx
+		}
+	}
+
+	for _, mb := range body.MiniBlocks {
+		numTxs += len(mb.TxHashes)
+
+		isCrossShardTxBlockFromSelf := mb.Type == block.TxBlock &&
+			mb.SenderShardID == tc.shardCoordinator.SelfId() &&
+			mb.ReceiverShardID != tc.shardCoordinator.SelfId()
+
+		if !isCrossShardTxBlockFromSelf {
+			continue
+		}
+
+		for txHash := range mb.TxHashes {
+			tx, ok := allTxs[string(txHash)]
+			if !ok {
+				log.Warn("transactionCoordinator.isMaxBlockSizeReached: tx not found",
+					"mb type", mb.Type,
+					"senderShardID", mb.SenderShardID,
+					"receiverShardID", mb.ReceiverShardID,
+					"numTxHashes", len(mb.TxHashes),
+					"tx hash", txHash)
+				continue
```

Comment:
why not escalate the error here, effectively stating that the block body is not ok?

Reply:
Actually this error could not happen, as the node can not reach in this point if it has not all the missing txs from the body received. But if there happened something with the source map I just want to signal with an Warn message and to continue the counting procedure even if we can not check if this missing tx is a sc call or not.

Reply:
It did due to programming errors :))